packages/core/src/components/DatePicker/DatePicker.tsx

```diff
@@ -115,46 +125,69 @@ export const DatePicker: React.FC<Props> & WithStyle = React.memo(
+            return (
+                <Wrapper
+                    id={`${id}-datepicker-wrapper`}
+                    ref={wrapperRef}
+                    fullWidth={fullWidth}
+                    minWidth={minWidth}
                     size={size}
-                    disabled={disabled}
-                    value={textValue}
-                    onChange={onTextChange}
-                    {...{ ...restProps, onBlur, onFocus, onInvalid }}
-                />
-                {showCalendar && (
-                    <Calendar
-                        id={`${id}-calendar`}
-                        date={date}
-                        isErrorPresent={isErrorPresent}
-                        onChange={onDateChange}
-                        minSelectableDate={minSelectableDate}
-                        maxSelectableDate={maxSelectableDate}
+                    className={className}
+                    placement={popoverPlacement}
+                >
+                    <TextField
+                        errorText={errorText || builtInErrorMessage}
+                        id={id}
+                        ref={inputRef}
+                        required={required}
+                        suffix={suffixEl}
+                        fullWidth
+                        mask={displayFormat.replace(new RegExp('\\/|\\-', 'g'), ' $& ').toUpperCase()}
+                        size={size}
+                        disabled={disabled}
+                        value={textValue}
+                        onChange={onTextChange}
+                        {...{ ...restProps, onBlur, onFocus, onInvalid }}
                     />
-                )}
-            </Wrapper>
-        );
+                    {showCalendar && (
+                        <Calendar
+                            id={`${id}-calendar`}
+                            date={date}
+                            isErrorPresent={isErrorPresent}
+                            onChange={onDateChange}
+                            minSelectableDate={minSelectableDate}
+                            maxSelectableDate={maxSelectableDate}
+                        />
+                    )}
+                </Wrapper>
+            );
+        } else {
+            return (
+                <Wrapper
+                    id={`${id}-datepicker-wrapper`}
+                    ref={wrapperRef}
+                    fullWidth={fullWidth}
+                    minWidth={minWidth}
+                    size={size}
+                    className={className}
+                    placement={popoverPlacement}
+                >
+                    <TextField
+                        onChange={handleOnChangeNative}
+                        type="date"
+                        placeholder="mm-dd-yyyy"
+                        suffix={suffixEl}
+                        errorText={errorText || builtInErrorMessage}
+                        fullWidth
+                        {...{ ...restProps, onBlur, onInvalid }}
```

Comment:
I see that this `TextField` has a lot less props than the other (such as `value`, `disabled`, and `required`) but I'm thinking it's intentional. Just curious about why the mobile version doesn't require them.

Reply:
Also we can add conditions on textField props only, no need to add whole wrapper component in else block

Reply:
https://github.com/medly/medly-components/pull/186/commits/42042fa56f33c374c281b5d1b6c37c157197f2aa 
Covers most conditions